rn/Teacher/src/modules/speedgrader/components/CanvadocViewer.js

```diff
@@ -33,9 +35,12 @@ export default class CanvadocViewer extends Component<Props> {
     return (
       this.props.style !== newProps.style ||
       this.props.config.drawerInset !== newProps.config.drawerInset ||
-      this.props.config.previewPath !== newProps.config.previewPath
+      this.props.config.previewPath !== newProps.config.previewPath ||
+      this.props.config.filename !== newProps.config.filename ||
+      this.props.config.fallbackURL !== newProps.config.fallbackURL
```

Comment:
this looks like the crux of the fix here. We weren't always re-rendering when you chose a different file.

Reply:
Correct. The other stuff with `isWide` was on iPad when you tapped a new file the file tab would deselect. Which made sense on iphone because the drawer would also close. But on ipad it would stay open and just take the user away from what they were tapping.